packages/rulesets/src/spectral/test/path-for-tracked-resource-types.test.ts

```diff
@@ -0,0 +1,670 @@
+              },
+            },
+          },
+        },
+        put: {
+          operationId: "Foo_Update",
+          description: "Test Description",
+          parameters: [
+            {
+              name: "foo_patch",
+              in: "body",
+              schema: {
+                $ref: "#/definitions/FooRequestParams",
+              },
+            },
+          ],
+          responses: {
+            "200": {
+              schema: {
+                $ref: "#/definitions/FooResource",
+              },
+            },
+          },
+        },
+      },
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachine/{vmName}/providers/Microsoft.Billing/extensions/{extensionName}":
+        {
+          get: {
+            operationId: "Foo_Update",
+            description: "Test Description",
+            parameters: [
+              {
+                name: "foo_patch",
+                in: "body",
+                schema: {
+                  $ref: "#/definitions/FooRequestParams",
+                },
+              },
+            ],
+            responses: {
+              "200": {
+                schema: {
+                  $ref: "#/definitions/FooResourceUpdate",
+                },
+              },
+            },
+          },
+          put: {
+            operationId: "Foo_Update",
+            description: "Test Description",
+            parameters: [
+              {
+                name: "foo_patch",
+                in: "body",
+                schema: {
+                  $ref: "#/definitions/FooRequestParams",
+                },
+              },
+            ],
+            responses: {
+              "200": {
+                schema: {
+                  $ref: "#/definitions/FooResource",
+                },
+              },
+            },
+          },
+        },
+    },
+    definitions: {
+      FooRequestParams: {
+        allOf: [
+          {
+            $ref: "#/definitions/FooProps",
+          },
+        ],
+      },
+      FooResource: {
+        allOf: [
+          {
+            $ref: "#/definitions/FooProps",
+          },
+        ],
+      },
+      FooResourceUpdate: {
+        allOf: [
+          {
+            $ref: "#/definitions/FooProps",
+          },
+        ],
+      },
+      FooProps: {
+        properties: {
+          location: {
+            type: "string",
+          },
+          tags: {
+            type: "object",
+            additionalProperties: {
+              type: "string",
+            },
+          },
+        },
+      },
+    },
+  }
+  return linter.run(oasDoc).then((results) => {
+    expect(results.length).toBe(0)
+  })
+})
+test("PathForTrackedResourceTypes should not find errors for other resources", () => {
+  const oasDoc = {
+    swagger: "2.0",
+    paths: {
+      "/subscriptions/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachine/{vmName}": {
```

Comment:
>"/subscriptions/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachine/{vmName [](http://example.com/codeflow?start=6&length=101)

wasnt subId mandatory?

Reply:
We are checking that scenario's in other rule in "verifyArmPath" function, this test is especially not to find errors if the resource is not tracked.